test/utils/Address.test.js

```diff
@@ -1,333 +1,287 @@
+    describe('with valid contract receiver', function () {
       it('calls the requested function', async function () {
-        const abiEncodedCall = this.target.contract.methods.mockFunction().encodeABI();
+        const abiEncodedCall = this.target.interface.encodeFunctionData('mockFunction');
+        const tx = await this.mock.$functionCall(this.target, abiEncodedCall);
 
-        const receipt = await this.mock.$functionCall(this.target.address, abiEncodedCall);
+        await expect(tx)
+          .to.emit(this.mock, 'return$functionCall')
+          .withArgs(coder.encode(['string'], ['0x1234']));
 
-        expectEvent(receipt, 'return$functionCall', {
-          ret0: web3.eth.abi.encodeParameters(['string'], ['0x1234']),
-        });
-        await expectEvent.inTransaction(receipt.tx, CallReceiverMock, 'MockFunctionCalled');
+        await expect(tx).to.emit(this.target, 'MockFunctionCalled');
       });
 
       it('calls the requested empty return function', async function () {
-        const abiEncodedCall = this.target.contract.methods.mockFunctionEmptyReturn().encodeABI();
-
-        const receipt = await this.mock.$functionCall(this.target.address, abiEncodedCall);
-
-        await expectEvent.inTransaction(receipt.tx, CallReceiverMock, 'MockFunctionCalled');
+        const abiEncodedCall = this.target.interface.encodeFunctionData('mockFunctionEmptyReturn');
+        await expect(this.mock.$functionCall(this.target, abiEncodedCall)).to.emit(this.target, 'MockFunctionCalled');
       });
 
       it('reverts when the called function reverts with no reason', async function () {
-        const abiEncodedCall = this.target.contract.methods.mockFunctionRevertsNoReason().encodeABI();
+        const abiEncodedCall = this.target.interface.encodeFunctionData('mockFunctionRevertsNoReason');
 
-        await expectRevertCustomError(
-          this.mock.$functionCall(this.target.address, abiEncodedCall),
+        await expect(this.mock.$functionCall(this.target, abiEncodedCall)).to.be.revertedWithCustomError(
+          this.mock,
           'FailedInnerCall',
-          [],
         );
       });
 
       it('reverts when the called function reverts, bubbling up the revert reason', async function () {
-        const abiEncodedCall = this.target.contract.methods.mockFunctionRevertsReason().encodeABI();
+        const abiEncodedCall = this.target.interface.encodeFunctionData('mockFunctionRevertsReason');
 
-        await expectRevert(this.mock.$functionCall(this.target.address, abiEncodedCall), 'CallReceiverMock: reverting');
+        await expect(this.mock.$functionCall(this.target, abiEncodedCall)).to.be.revertedWith(
+          'CallReceiverMock: reverting',
+        );
       });
 
       it('reverts when the called function runs out of gas', async function () {
-        const abiEncodedCall = this.target.contract.methods.mockFunctionOutOfGas().encodeABI();
+        if (process.env.COVERAGE) this.skip();
```

Comment:
Not elegant but this test was not working in coverage

Reply:
I think that is because you were not using `gasLimit` to set the transaction's amount of gas. This is a big difference between web3.js and ethers.js

Reply:
Also, when a tests needs to be disable on coverage, we name it
```javascript
it('[skip-on-coverage] ...', async function () {
   ...
});
```

(that prefix is specified in `.solcover.js`)

Reply:
Can't believe I missed this

> I think that is because you were not using `gasLimit` to set the transaction's amount of gas. This is a big difference between web3.js and ethers.js

Can't believe I missed this